lib/reek/configuration/configuration_file_finder.rb

```diff
@@ -17,42 +17,102 @@ class ConfigFileException < StandardError; end
     # The order in which ConfigurationFileFinder tries to find such a
     # configuration file is exactly like above.
     module ConfigurationFileFinder
-      module_function
```

Comment:
Maybe we should settle on either `class << self` or `def self.something_something`. I've a slight preference for the latter, it also seems to be the method used most in Reek, and it would mean a nicer diff here :smile:. WDYT?